### PR TITLE
Clipboard import

### PR DESCRIPTION
**Pull request by Cha0sX**
Detect trojan url from Clipboard and fill into the EditText.

### CODE REVIEW

File: app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java [hunk 1]
```diff
@@ -342,6 +351,44 @@ public void onRun() {
...
+                            public void onClick(DialogInterface dialog, int which) {
+                                trojanURLText.setText(clipboardText);
+                            }
+                        })
+                        .setNegativeButton(R.string.common_cancel, null)
+                        .create()
+                        .show();
+            }
+        });
+    }
```
Comment (TchaikovDriver) [inline app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java:390]: Hi Cha0sX, if you just want to use `Handler` for inserting the action at the rear of events queue, I suggest calling `Threads.instance().runOnUiThread(Runnable)` instead.

Comment (Cha0sX) [reply app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java:391]: Done👌

**Comment by wongsyrone** [comment]
@TchaikovDriver  Please review.

**Comment by wongsyrone** [comment]
Please use https://github.com/trojan-gfw/igniter/blob/75557b993a0e715535be1a5ca3bd51625a97d268/app/src/main/java/io/github/trojan_gfw/igniter/TrojanURLHelper.java#L23 to validate the trojan URL

File: app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java [hunk 2]
```diff
@@ -355,8 +353,11 @@ public void onRun() {
     protected void onResume() {
         super.onResume();
 
-        // use handler to getPrimaryClip
-        mHandler.post(new Runnable() {
+        detectClipboard();
```
Comment (wongsyrone) [inline app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java:356]: Its naming is too vague, how about 'CheckTrojanURLFromClipboard'?

Comment (Cha0sX) [reply app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java:356]: 😂It's better

**Comment by Cha0sX** [comment]
> Please use
> 
> https://github.com/trojan-gfw/igniter/blob/75557b993a0e715535be1a5ca3bd51625a97d268/app/src/main/java/io/github/trojan_gfw/igniter/TrojanURLHelper.java#L23
> 
> to validate the trojan URL

good Helper, all done

**Comment by TchaikovDriver** [comment]
> @TchaikovDriver Please review.

LGTM.

**Comment by Cha0sX** [comment]
> > @TchaikovDriver Please review.
> 
> It seems that it failed in CircleCI tests.

The reason is:
Task 'crashlyticsUploadDistributionDevelopmentRelease' not found in root project 'code'.

**Comment by TchaikovDriver** [comment]
> > > @TchaikovDriver Please review.
> > 
> > 
> > It seems that it failed in CircleCI tests.
> 
> The reason is:
> Task 'crashlyticsUploadDistributionDevelopmentRelease' not found in root project 'code'.

I ran the latest code and found an exception about this commit:
```
Process: io.github.trojan_gfw.igniter, PID: 13505
    java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' on a null object reference
        at io.github.trojan_gfw.igniter.MainActivity$14.run(MainActivity.java:371)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6854)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
```
The variable `clipboardText` is null when I enter Igniter with empty clipboard. Then it throws a `NullPointerException` at `clipboardText.toString()`.

**Comment by Cha0sX** [comment]
> > > > @TchaikovDriver Please review.
> > > 
> > > 
> > > It seems that it failed in CircleCI tests.
> > 
> > 
> > The reason is:
> > Task 'crashlyticsUploadDistributionDevelopmentRelease' not found in root project 'code'.
> 
> I ran the latest code and found an exception about this commit:
> 
> ```
> Process: io.github.trojan_gfw.igniter, PID: 13505
>     java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' on a null object reference
>         at io.github.trojan_gfw.igniter.MainActivity$14.run(MainActivity.java:371)
>         at android.os.Handler.handleCallback(Handler.java:873)
>         at android.os.Handler.dispatchMessage(Handler.java:99)
>         at android.os.Looper.loop(Looper.java:193)
>         at android.app.ActivityThread.main(ActivityThread.java:6854)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:860)
> ```
> 
> The variable `clipboardText` is null when I enter Igniter with empty clipboard. Then it throws a `NullPointerException` at `clipboardText.toString()`.

I cant reproduce this issue in my situation. It can be solved simply with a null check, but I think its not the actual reason in this case, can you debug and evaluate the code  `mClipboardManager.getPrimaryClip()` to see what's in the ClipData?

**Pull request merged**